codegen-on-oss/codegen_on_oss/analysis/analysis.py

```
@@ -62,18 +70,34 @@ def get_monthly_commits(repo_path: str) -> Dict[str, int]:
         original_dir = os.getcwd()
 
         with tempfile.TemporaryDirectory() as temp_dir:
-            subprocess.run(["git", "clone", repo_path, temp_dir], check=True)
+            # Use full path to git executable
+            git_executable = shutil.which("git")
+            if not git_executable:
+                raise GitExecutableNotFoundError()
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)